### ISSUE TITLE
IOException while building

### ISSUE THREAD

**Issue by Martin-Wegner** [ktoso/maven-git-commit-id-plugin#203]
Since 2.1.14 or 2.1.15 I get this error:

```java
java.io.IOException: Cannot run program "bash" (in directory "C:\Users\user"): CreateProcess error=2, Das System kann die angegebene Datei nicht finden
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at java.lang.Runtime.exec(Runtime.java:620)
	at org.eclipse.jgit.util.FS.readPipe(FS.java:431)
	at org.eclipse.jgit.util.FS_Win32.discoverGitPrefix(FS_Win32.java:113)
	at org.eclipse.jgit.util.FS.gitPrefix(FS.java:517)
	at org.eclipse.jgit.util.SystemReader$Default.openSystemConfig(SystemReader.java:92)
	at org.eclipse.jgit.internal.storage.file.FileRepository.<init>(FileRepository.java:171)
	at org.eclipse.jgit.storage.file.FileRepositoryBuilder.build(FileRepositoryBuilder.java:92)
	at pl.project13.maven.git.JGitProvider.getGitRepository(JGitProvider.java:265)
	at pl.project13.maven.git.JGitProvider.init(JGitProvider.java:79)
	at pl.project13.maven.git.GitDataProvider.loadGitData(GitDataProvider.java:91)
	at pl.project13.maven.git.GitCommitIdMojo.loadGitDataWithJGit(GitCommitIdMojo.java:585)
	at pl.project13.maven.git.GitCommitIdMojo.loadGitData(GitCommitIdMojo.java:558)
	at pl.project13.maven.git.GitCommitIdMojo.execute(GitCommitIdMojo.java:376)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:189)
	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:185)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: CreateProcess error=2, Das System kann die angegebene Datei nicht finden
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 26 more
```

The error should be fixed in jgit 4.0.0.201506090130-r:
https://github.com/eclipse/jgit/commit/ec6ec3b10fb1ef8dd73a499d0b1f7a7d711b84dd

So I think you should update the maven-git-commit-id-plugin jgit pom dependency?

**Comment by ktoso** [comment]
Would be cool if you could try updating the dependency and run locally, so we'll know if it works.
I don't have a windows box at my disposal.

Vielen dank im vorhinein Martin!

**Comment by Martin-Wegner** [comment]
It works but you must change JGitCommon.java line 220 from walk.release(); to walk.close();

After that the following unit tests will fail:

GitCommitIdMojoIntegrationTest:
->  shouldFailWithExceptionWhenNoGitRepoFound (because the directory is found)
->  shouldSkipWithoutFailOnNoGitDirectoryWhenNoGitRepoFound (because the directory is found)

GitCommitIdMojoTest
-> testCraftPropertiesOutputFileWithRelativePath (because you use "/" instead of File.separator or System.lineSeparator())
-> testCraftPropertiesOutputFileWithFullPath (because you use "/" instead of File.separator or System.lineSeparator())

**Comment by ktoso** [comment]
Sounds like legit mistakes in tests, thanks for investigating!

**Comment by TheSnoozer** [comment]
fixed by #205. Thanks!

**Comment by TheSnoozer** [comment]
Workaround:
https://github.com/UniversalMediaServer/UniversalMediaServer/issues/627#issuecomment-126943376
